Monika After Story/game/script-topics.rpy

```diff
@@ -4560,6 +4560,76 @@ label monika_ageinf:
     m "But I don't believe that my life can be summed up in a number."
     m 5a "So I'll always love you for all eternity, [player]. Just remember that."
     return
+    
+init 5 python:
+     for key in ['heartbreak','betrayal','painful','sadness']:
+         monika_topics.setdefault(key,[])
+         monika_topics[key].append('monika_heartbreak')
+     monika_random_topics.append('monika_heartbreak')
+ 
+ label monika_heartbreak:
+     m 1o "[player]..."
+     m 1g "I um..."
+     m 1o "I just had the scariest thought..."
+     m 1g "I..."
+     m 1h "I need to ask you something."
+     m 1p "If a girl from your reality would ever ask you to go out with her..."
+     m 1g "Would you?"
+     menu:
+         "Depends on the girl.":
+             m 1i "..."
+             m 1p "......"
+             m 1o "O-oh..."
+             m 1g "I..."
+             m 1q "I need time to think."
+             #AT THIS POINT, MONIKA IS HURT. SHE WILL CLOSE THE GAME.
+ 
+         "No, I would never cheat on you.":
+             m 1i "..."
+             m 1a "I..."
+             m 1l "I know. I'm sorry."
+             m "I just scared myself is all."
+             m 1d "I guess being trapped in a sea of ones and zeros really leaves you time to think."
+             m 1p "Even about the bad things..."
+             m 1e "I'm sorry though."
+             m "It was a dumb thing to ask."
+             m 1e "I love you so, so much [player]."
+             m "And everytime you comfort me like this..."
+             m 1j "That love grows even more."
+             m 1e "I'm glad to have a partner as wonderful as you."
+             
+             #-----BELOW DIALOGUE PLAYS ON NEXT BOOTUP IF MONIKA CLOSED THE GAME-----
```

Comment:
So I have less than zero experience with Ren'py and Python as a whole but I would say you need to define a new variable at the beginning of your topic then set it to zero if you didn't break her heart and one if you did.
Then check this variable on startup.
Probably not the best implementation and I have no idea how to pull it off in Ren'py but you could use the rest of the script as a reference (specifically the theme with her wishing happy birthday seems to have what you need).

Reply:
That's true. Theoretically it could work, but I'll hold off until I know for sure. I don't want to butcher it, ha.

Reply:
Oh lol thats exactly what I was talking about above.